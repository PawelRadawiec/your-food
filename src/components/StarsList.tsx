import React, { useEffect, useState } from 'react';
import { FlatList, StyleSheet } from 'react-native';
import { MaterialIcons } from '@expo/vector-icons';

const StarsList = ({ rating }: { rating: number }) => {
  const [stars, setStars] = useState<number[]>([]);
  const [isDecimal, setIsDecimal] = useState<boolean>(false);
  useEffect(() => {
    setIsDecimal(rating % 1 !== 0);
    const stars: number[] = [];
    for (let i = 0; i < Math.floor(rating); i++) {
      stars.push(i);
    }
    setStars(stars);
  }, [rating]);
  return (
    <FlatList
      horizontal={true}
      data={stars}
      keyExtractor={(item) => `id${item}`}
      renderItem={({ index }) => {
        return isDecimal && index === stars.length - 1 ? (
          <MaterialIcons name="star-half" size={24} color="black" />
        ) : (
          <MaterialIcons name="star" size={24} color="black" />
        );
      }}
    />
  );
};

const styles = StyleSheet.create({});

export default StarsList;
