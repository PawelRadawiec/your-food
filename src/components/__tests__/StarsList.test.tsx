import React from 'react';
import { shallow, mount } from 'enzyme';
import StarsList from '../StarsList';
import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { FlatList } from 'react-native';
import { MaterialIcons } from '@expo/vector-icons';

Enzyme.configure({ adapter: new Adapter() });

describe('Stars list unit test', () => {
  it('should render my component', () => {
    const component = shallow(<StarsList rating={5} />);
    expect(component.getElements()).toMatchSnapshot();
  });

  it('should contains 5 icons', () => {
    const component = mount(<StarsList rating={5} />);
    const flatList = component.find(FlatList);
    const icons = flatList.find(MaterialIcons);
    expect(icons.length).toBe(5);
  });

  it('should contains 4 icons for 4.5 rating', () => {
    const component = mount(<StarsList rating={4.5} />);
    const flatList = component.find(FlatList);
    const icons = flatList.find(MaterialIcons);
    expect(icons.length).toBe(4);
  });

});
