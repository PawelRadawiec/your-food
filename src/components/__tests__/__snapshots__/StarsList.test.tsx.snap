// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stars list unit test should render my component 1`] = `
Array [
  <FlatList
    data={Array []}
    disableVirtualization={false}
    horizontal={true}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onEndReachedThreshold={2}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  />,
]
`;
